lib/mcb/courses_editor.rb

```diff
@@ -118,16 +142,42 @@ def print_at_most_two_courses
     end
 
     def print_existing(attribute_name)
-      puts "Existing values for course #{attribute_name}:"
-      table = Tabulo::Table.new @courses.order(:course_code) do |t|
-        t.add_column(:course_code, header: "course\ncode", width: 4)
-        t.add_column(attribute_name) unless attribute_name == :course_code
+      # don't print the existing attributes when creating a new course, since
+      # this will be null in the majority of cases
+      if @courses.select(&:persisted?).all?
+        puts "Existing values for course #{attribute_name}:"
+        table = Tabulo::Table.new @courses do |t|
+          t.add_column(:course_code, header: "course\ncode", width: 4)
+          t.add_column(attribute_name) unless attribute_name == :course_code
+        end
+        puts table.pack(max_table_width: nil), table.horizontal_rule
+      end
+    end
+
+    def print_summary
+      puts "\nHere's the final course that's been created:"
+      print_at_most_two_courses
+      @cli.enter_to_continue
+    end
+
+    def try_saving_course
+      if course.valid?
+        puts "Saving the course"
+        course.save!
+      else
+        puts "Course isn't valid:"
+        course.errors.full_messages.each { |error| puts " - #{error}" }
```

Comment:
If this fails, the user will go straight to editing subjects and sites, which will both fail, which will fail, I think, since `course` isn't valid. Would it make more sense to kick them back up to the top of the wizard to try again before proceeding?

Reply:
I've added 30c563c, which does the slightly simpler thing of aborting the wizard if the course fails validation. From that point onward, it's easy for the support agent to just restart the process by invoking the previous command and copying their previous answers from the terminal history.